src/content/agenda.astro

```diff
@@ -1,45 +1,108 @@
 ---
-import { agenda } from "../data/agenda";
-import { speakers } from "../data/speakers";
+import type {
+  AgendaItem,
+  PersonalDetails,
+  Speaker,
+  TalkDetails,
+} from "../data/types";
+import { getPersonIdFromFilePath } from "../utils";
+
+export interface Props {
+  agendaItems: AgendaItem[];
+  title: string;
+}
+
+const speakers = await Astro.glob<PersonalDetails>("../data/speakers/*.md");
+const speakerTalks = await Astro.glob<TalkDetails>(
+  "../data/speaker-talks/*.md"
+);
+
+const { agendaItems, title } = Astro.props as Props;
+
+const getSpeakerInfo = (speakerId: string): Speaker | null => {
+  const speakerDetails = speakers.find(
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  const speakerTalk = speakerTalks.find(
+    (s) => speakerId === getPersonIdFromFilePath(s.file)
+  );
+
+  if (!speakerDetails || !speakerTalk) {
+    return null;
+  }
```

Comment:
I believe you only need to check for `!speakerDetails` considering you're using a fallback in the markup

Reply:
Yes indeed, but I prefer to use `null` explicitly for marking the absence of a value. The conditional chaining operator will return a bunch of `undefined` values